### PR TITLE
Add `EncodeAsFields` impls for Arc/Rc/Box

### LINKED ISSUE #18: Implement `EncodeAsFields` for pointer types like `Arc` and `Box`
This issue indicates that it's not implemented for `Arc`:

https://github.com/paritytech/subxt/issues/1241

Let's add impls (for at least `Box`, `Arc`, and `Rc`.. anything else?) and check that they work in Subxt.

This could possible be in a patch release.

### PR DESCRIPTION
**Pull request by pkhry**
### Description: 
Close #18

### CODE REVIEW

File: scale-encode/src/impls/mod.rs [hunk 1]
```diff
@@ -1215,4 +1235,65 @@ mod test {
             123u64,
         );
     }
+
+    #[test]
+    fn smart_pointer_fields() {
```
Comment (jsdw) [inline scale-encode/src/impls/mod.rs]: Nit: clarify the name a touch? :)

```suggestion
    fn encode_smart_pointers_as_fields() {
```

**Review by jsdw — APPROVED** [review]
LGTM!

**Review by lexnv — APPROVED** [review]
Nice one!

**Comment by niklasad1** [comment]
@pkhry https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword,  in the future just write "close, fix or resolves #xx` in the PR description to automatically link to the issue :)

**Review by niklasad1 — APPROVED** [review]
Looks good :)

**Pull request merged**